### DISCUSSION TITLE
Valetudo RE 0.10.8

### DISCUSSION THREAD

**Discussion by rand256** [rand256/valetudo#489] [General]
How are you doing? This is the service update for 0.10 RE release, verified on S55 (fw 2034).
Changes in this version:

* Expected cleaning direction in a zone is now showed on the map with the arrows sign;
* Updated **"go_to"** and **"zoned_cleanup"** MQTT commands to allow specifying their targets directly with coordinates (predefined zones or spots aren't mandatory anymore);
* Added **"remote_control"** MQTT command (and its REST API synonym) to automate manual remote controlling the device. With this you can send a sequence of such commands for the device to automatically chain them, and it will automatically start and end manual control mode when required;
* Double tap on the map will now show coordinates of tapped position;
* Node.js updated to 16.16 and some dependencies were updated too.

Note that Node.js 16 binary this time is a default build from pkg-fetch repo. This primarily means that it is rather big compared to my previous custom Node.js 12 builds and there's a chance that some CPU optimizations are not enabled here. While it doesn't sound awesome, but still you'll hardly notice any substantial differences I believe.

This release includes a pre-build image for Gen2 based on firmware version 2034 and an image for Gen1 based on firmware version 4028. Both are built by vacuumz image builder and use password "cleaner" for root login at SSH by default. You are advised to change it right after install - you know it for sure.

If you already have a previous release installed, you can install only RE update using a deb file (for gen1) or via [this guide](https://github.com/rand256/valetudo/wiki/Updating-valetudo-binary-on-2008-firmware) (for gen2 and 3).

<hr /><em>This discussion was created from the release <a href='https://github.com/rand256/valetudo/releases/tag/0.10.8'>Valetudo RE 0.10.8</a>.</em>

**Comment by GitMonzter** [comment]
Thanks for continuing on this fine project! Experienced this:
S50 (Gen.2) updates fine :-)

while S5(Gen.1) does not update :-(
After update trial, old map data cannot be accessed anymore ("operation timed out").
Cleaning start is possible but no new map data is generated. Always showing "NO MAP DATA".

So, trying to update seems to damage part of the mapping.
Cleaning did not succeed, strangely. Will do more tests by tomorrow.
Re-Install does 0.10.7 did not fix it.

**So, for the moment, I would not recommend S5 (Gen.1) users to try updating.**

Hope this helps to find the bug ....
Ideas what to do....?

**Reply by rand256** [reply]
The only difference between S50 and S55 is color: S50 is white and S55 is black. That's it.
Despite colors the devices are the same.

"NO MAP DATA" can be seen for a while after any reboot when device attempts to send map to disconnected cloud instead of its local interface, waiting usually helps.

**Reply by GitMonzter** [reply]
Sorry, I mixed up he names - corrected it ...

**Reply by rand256** [reply]
Did you install the full image or just replaced valetudo binary?

You can probably try to factory reset and reinstall either 0.10.7 or 0.10.8 just to be sure, but I must say that both versions were built using exactly the same image builders with exactly the same settings, while the most significant difference in between is only in Node.js base binary updated from v12 to v16. Therefore if you can open valetudo web UI then it's assumed that binary works fine.

**Reply by GitMonzter** [reply]
Full image and web interface works.

**Reply by GitMonzter** [reply]
A lot of these in log:
2022-08-25T19:48:58.154Z Failed to get response for message: miIO.get_ota_state [] { retries: 25, retriesHS: 29 }
2022-08-25T19:48:58.147Z Probed last id = 700950 using get_status (2 retries)

**Reply by GitMonzter** [reply]
Will check more functionality by tomorrow.

**Reply by GitMonzter** [reply]
Giving the robos a night time, they work normal again (on 0.10.7) - Hooray! :-)
Thanks a lot for your lightning quick answer.

**Reply by rand256** [reply]
I guess it would also work as intended on 0.10.8 too. No reason why it shouldn't.
